topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,415 @@
-3. Run the application on Android to ensure that everything works correctly.
+   ![Greeting from the KMM module](xcode-iphone-hello.png){width=300}
 
-### Make your application work on iOS
+3. In `ContentView.swift`, write [code for using data from the KMM module and rendering the application UI](https://github.com/Kotlin/kmm-integration-sample/blob/final/SimpleLoginIOS/SimpleLoginIOS/ContentView.swift).
 
-For `expect` declarations in the shared code, add the required [`actual` implementations for iOS](connect-to-platform-specific-apis.md).
+4. In `simpleLoginIOSApp.swift`, import the `shared` module and specify the arguments for the `ContentView()` function:
 
-If you don't have an iOS application, create an Xcode project and specify the path to it in `gradle.properties`. 
-If you already have an Xcode project, simply specify a relative or absolute path to the project.
+    ```Swift
+    import SwiftUI
+    import shared
+    
+    @main
+    struct SimpleLoginIOSApp: App {
+        var body: some Scene {
+            WindowGroup {
+                ContentView(viewModel: .init(loginRepository: LoginRepository(dataSource: LoginDataSource()), loginValidator: LoginDataValidator()))
+            }
+        }
+    }
+   ```  
 
-```kotlin
-xcodeproj=./iosApp
-```
+   ![Simple login application](xcode-iphone-login.png){width=300}
+
+## Enjoy the results – update the logic only once
+
+Now your application is cross-platform. You can update the business logic in one place and see results on both Android and iOS.
+
+1. In Android Studio, change the validation logic for a user's password in the `checkPassword()` function of the `LoginDataValidator` class:
+
+    ```kotlin 
+    package com.jetbrains.simplelogin.shared.data
+
+    class LoginDataValidator {
+    //... 
+        fun checkPassword(password: String): Result {
+            return when  {
+                password.length < 5 -> Result.Error("Password must be >5 characters")
+                password.toLowerCase() == "password" -> Result.Error("Password shouldn't be \"password\"")
+                else -> Result.Success
+            }
+    }
+    //...
+    }
+    ``` 
+
+2. Update `gradle.properties` to connect your iOS application to Android Studio for running it on a simulated or real device right there:
+
+    ```text
+    xcodeproj=iosApp/SimpleLoginIOS.xcodeproj
+    ```
+
+3. Apply the changes by synchronizing the project.
+
+   ![Synchronize the Gradle files](gradle-sync.png)
+
+You will see the new run configuration **simpleLoginIOS** for running your iOS application right from Android Studio.
+
+![iOS run configuration](ios-run-configuration-simplelogin.png){width=200}
+
+![iOS application password error](iphone-password-error.png){width=300}
 
-Once you're done, run your application on iOS to ensure that it works correctly.  
+![Android application password error](android-password-error.png){width=300}
```

Comment:
minor: Maybe combine three pictures into one (putting phones side by side) to save space?

Reply:
I thought about this. Maybe combine or create a table. Need to check what works better on different devices. But I'd do this consistently in other KMM docs as well. We'll discuss with Vika.